docs/CustomPipelineGraphicsMod.md

```diff
@@ -0,0 +1,251 @@
+|``texmap_to_texcoord_index`` | uint[]                  | v1    | An array of texture units to texture coordinate values                                        |
+|``lights_chan0_color``       | CustomShaderLightData[] | v1    | An array of color lights for channel 0, the amount is specified by ``light_chan0_color_count``|
+|``lights_chan0_alpha``       | CustomShaderLightData[] | v1    | An array of alpha lights for channel 0, the amount is specified by ``light_chan0_alpha_count``|
+|``lights_chan1_color``       | CustomShaderLightData[] | v1    | An array of color lights for channel 1, the amount is specified by ``light_chan1_color_count``|
+|``lights_chan1_alpha``       | CustomShaderLightData[] | v1    | An array of alpha lights for channel 1, the amount is specified by ``light_chan1_alpha_count``|
+|``ambient_lighting``         | vec4[]                  | v1    | An array of ambient lighting values.  Count is two, one for each color channel                |
+|``base_material``            | vec4[]                  | v1    | An array of the base material values.  Count is two, one for each color channel               |
+|``tev_stages``               | CustomShaderTevStage[]  | v1    | An array of TEV stages, the amount is specified by ``tev_stage_count``                        |
+|``tev_stage_count``          | uint                    | v1    | The count of TEV stages                                                                       |
+|``final_color``              | vec4                    | v1    | The final color generated by Dolphin after all TEV stages are executed                        |
+|``time_ms``                  | uint                    | v1    | The time that has passed in milliseconds, since the game was started.  Useful for animating   |
+
+`CustomShaderLightData` is used to denote lighting data the game is applying when rendering the specific draw call.  It has the following structure:
+
+|Name                     | Type                    | Since | Description                                                                                     |
+|-------------------------|-------------------------|-------|-------------------------------------------------------------------------------------------------|
+|``position``             | vec3                    | v1    | The position of the light in _view space_                                                       |
+|``direction``            | vec3                    | v1    | The direction in _view space_ the light is pointing (only applicable for point and spot lights) |
+|``color``                | vec3                    | v1    | The color of the light                                                                          |
+|``light_type``           | uint                    | v1    | Whether the light is a color or alpha light.  See details below                                 |
```

Comment:
I'm pretty sure you deleted

```markdown
|``attenuation_type``     | uint                    | v1    | The attentuation type of the light.  See details below                                          |
```

when you meant to delete this line, as `light_type` no longer exists but `attenuation_type` does.

Reply:
Dang, I did.  Thanks! :)

Reply:
For reference on how I noticed this:

1. `git fetch origin 5cf812458a162bd91b18f385266d1d4eba9c9f65` to get the old version of your PR
2. `git switch -d 5cf812458a162bd91b18f385266d1d4eba9c9f65` to switch to it
3. `git fetch origin 91c8cc9e4b70784a69d8967e8662a17a4f979fc0` to get the new version of your PR
4. `git log 91c8cc9e4b70784a69d8967e8662a17a4f979fc0` to find the commit from master your PR was based on (I'm sure there's an automatic way to do that, but I don't know what it is). In this case it was 1854ff01422a69bde7b2267e9ed4fc69c30ee37a.
5. `git rebase 1854ff01422a69bde7b2267e9ed4fc69c30ee37a` to rebase the old version of the PR (as a detached HEAD)
6. `git diff HEAD 91c8cc9e4b70784a69d8967e8662a17a4f979fc0` to compare the old one with the new one.

It'd be nice if github's compare changes functionality automatically handled this, but instead you also get changes in master that aren't relevant.

Reply:
I think there's some way (locally?) to do this with the ref log but I've never really looked into it.  Nice sleuthing!